test/typescript.test.js

```diff
@@ -0,0 +1,10 @@
+import test from "tape";
```

Comment:
@bdwain here i followed the pattern used by redux for testing their type declarations. seemed safe to introduce the same dependencies as them + beneficial to have some tests for the purposes of refining the type system in less of a vacuum

Reply:
sweet looks awesome!